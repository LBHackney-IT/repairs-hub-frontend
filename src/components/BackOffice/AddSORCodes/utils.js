import { frontEndApiRequest } from '@/root/src/utils/frontEndApiClient/requests'

export const fetchContractors = () =>
  new Promise(async (resolve) => {
    const contractors = await frontEndApiRequest({
      method: 'get',
      path: '/api/contractors?getAllContractors=true',
    })

    resolve(contractors)
  })

export const fetchTrades = () =>
  new Promise(async (resolve) => {
    const trades = await frontEndApiRequest({
      method: 'get',
      path: '/api/schedule-of-rates/trades?getAllTrades=true',
    })

    resolve(trades)
  })

export const fetchContracts = (contractorReference) =>
  new Promise(async (resolve) => {
    const contracts = await frontEndApiRequest({
      method: 'get',
      path: `/api/backoffice/contracts?contractorReference=${contractorReference}`,
    })

    resolve(contracts)
  })

export const saveSorCodesToDatabase = async (data) => {
  await frontEndApiRequest({
    method: 'post',
    path: `/api/backoffice/sor-codes`,
    requestData: data,
  })
}

export const dataToRequestObject = (csvArray, contractReference, tradeCode) => {
  const sorCodes = csvArray.map((x) => ({
    code: x.Code,
    cost: parseFloat(x.Cost),
    standardMinuteValue: parseFloat(x.StandardMinuteValue),
    shortDescription: x.ShortDescription,
    longDescription: x.LongDescription,
  }))

  return {
    contractReference: contractReference,
    tradeCode: tradeCode,
    sorCodes: sorCodes,
    contract: 'placeholder',
  }
}
