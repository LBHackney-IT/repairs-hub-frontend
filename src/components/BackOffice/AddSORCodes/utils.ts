import { frontEndApiRequest } from '@/root/src/utils/frontEndApiClient/requests'

import { Contractor, Trade, NewSorCodesRequestObject } from './types'

export const fetchContractors = async () : Promise<Array<Contractor>>  => {
  return await frontEndApiRequest({
    method: 'get',
    path: '/api/contractors?getAllContractors=true',
    params: null,
    paramsSerializer: null,
    requestData: null
  })
}

export const fetchTrades = async () : Promise<Array<Trade>> => {
  return await frontEndApiRequest({
    method: 'get',
    path: '/api/schedule-of-rates/trades?getAllTrades=true',
    params: null,
    paramsSerializer: null,
    requestData: null
  })
}

export const fetchContracts = async (contractorReference: string) : Promise<Array<string>> => {
  return await frontEndApiRequest({
    method: 'get',
    path: `/api/backoffice/contracts?contractorReference=${contractorReference}`,
    params: null,
    paramsSerializer: null,
    requestData: null
  })
}

export const saveSorCodesToDatabase = async (data: NewSorCodesRequestObject): Promise<void> => {
  await frontEndApiRequest({
    method: 'post',
    path: `/api/backoffice/sor-codes`,
    requestData: data,
    params: null,
    paramsSerializer: null,
  })
}

export const dataToRequestObject = (csvArray: Array<any>, contractReference: string, tradeCode: string): NewSorCodesRequestObject => {
  const sorCodes = csvArray.map((x) => ({
    code: x.Code,
    cost: parseFloat(x.Cost),
    standardMinuteValue: parseFloat(x.StandardMinuteValue),
    shortDescription: x.ShortDescription,
    longDescription: x.LongDescription,
  }))

  return {
    contractReference: contractReference,
    tradeCode: tradeCode,
    sorCodes: sorCodes,
  }
}
