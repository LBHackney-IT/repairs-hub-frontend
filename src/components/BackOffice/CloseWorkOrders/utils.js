export const formatWorkOrderReferences = (workOrderReferences) => {
  return [
    ...new Set(
      workOrderReferences
        .split('\n')
        .map((x) => x.trim())
        .filter((x) => x)
    ),
  ]
}

const validateWorkOrderReference = (workOrderReference) => {
  const regex = /^\d{8}$/gm

  return regex.exec(workOrderReference) !== null
}

export const getInvalidWorkOrderReferences = (workOrderReferences) => {
  return workOrderReferences
    .filter((x) => !validateWorkOrderReference(x))
    .map((x) => `"${x}"`)
}

export const formatInvalidWorkOrderReferencesError = (
  invalidWorkOrderReferences
) => {
  return `Invalid WorkOrder Reference(s) entered: ${invalidWorkOrderReferences.join(
    ', '
  )}`
}

export const dateIsInFuture = (date) => {
  const today = new Date()

  return date > today
}
