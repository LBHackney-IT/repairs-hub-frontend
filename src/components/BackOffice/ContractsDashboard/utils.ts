import Contract from '@/root/src/models/contract'

export const today = new Date()

export function monthsOffset(months: number) {
  return new Date(
    today.getFullYear(),
    today.getMonth() + months,
    today.getDate()
  )
}

export const filterContractsByExpiryDate = (
  contracts: Contract[],
  months: number
) => {
  const contractExpiryCutOffDate = monthsOffset(months)

  const filteredContracts = contracts?.filter((contract) => {
    return (
      new Date(contract.terminationDate) > today &&
      new Date(contract.terminationDate) < contractExpiryCutOffDate
    )
  })
  return filteredContracts
}

export const filterRelativeInactiveContracts = (
  contracts: Contract[],
  yearAfter: string
) => {
  const relativeInactiveContracts = contracts?.filter(
    (contract) =>
      contract.terminationDate > yearAfter &&
      contract.terminationDate < new Date().toISOString()
  )
  return relativeInactiveContracts
}
