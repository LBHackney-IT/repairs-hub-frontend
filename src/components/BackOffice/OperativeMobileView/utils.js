export const filterOperatives = (operatives, operativeFilter) => {
  if (!operativeFilter) return operatives;

  return operatives.filter((x) =>
    x.id?.toString().includes(operativeFilter) ||
    x.name?.toLowerCase().includes(operativeFilter) ||
    x.payrollNumber?.toLowerCase().includes(operativeFilter)
  );
}
