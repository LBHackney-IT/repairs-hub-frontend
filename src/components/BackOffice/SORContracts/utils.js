export const sanitizeInput = value => {
    return value.trim().replaceAll(' ', '')
}

export const saveContractChangesToDatabase = async (data) => {
    await frontEndApiRequest({
        method: 'POST',
        path: `/api/backOffice/sor-contracts`,
        requestData: data,
      })
  }