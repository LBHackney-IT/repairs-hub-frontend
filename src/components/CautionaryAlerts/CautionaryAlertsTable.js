import PropTypes from 'prop-types'
import { Table, THead, TBody, TR, TH, TD } from '../Layout/Table'

const CautionaryAlertsTable = ({ cautionaryAlerts, query }) => {
  return (
    <>
      <Table>
        <THead>
          <TR className="lbh-body">
            <TH scope="col">Code</TH>
            <TH scope="col">Description</TH>
          </TR>
        </THead>
        <TBody>
          {cautionaryAlerts.map((cautionaryAlert, index) => (
            <TR
              key={index}
              index={index}
            >
              {query && query.includes(cautionaryAlert.code) ? (
                <>
                  <TD className="text-dark-red lbh-!-font-weight-bold">
                    {cautionaryAlert.code}
                  </TD>
                  <TD className="text-dark-red lbh-!-font-weight-bold">
                    {cautionaryAlert.description}
                  </TD>
                </>
              ) : (
                <>
                  <TD>{cautionaryAlert.code}</TD>
                  <TD>{cautionaryAlert.description}</TD>
                </>
              )}
            </TR>
          ))}
        </TBody>
      </Table>
    </>
  )
}

CautionaryAlertsTable.propTypes = {
  cautionaryAlerts: PropTypes.arrayOf(
    PropTypes.shape({
      code: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
  query: PropTypes.array,
}

export default CautionaryAlertsTable
