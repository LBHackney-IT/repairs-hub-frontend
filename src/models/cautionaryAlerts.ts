export type CautionaryAlert = {
  type: string
  comments: string
  startDate: string
  endDate: string
}

export type CautionaryAlertsResponse = {
  reference: string
  alerts: CautionaryAlert[]
}
