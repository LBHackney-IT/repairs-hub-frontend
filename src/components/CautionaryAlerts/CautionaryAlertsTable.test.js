import { render } from '@testing-library/react'
import UserContext from '../UserContext'
import CautionaryAlertsTable from './CautionaryAlertsTable'
import { operative } from 'factories/operative'
import { CAUTIONARY_ALERTS } from '@/utils/cautionaryAlerts'

describe('CautionaryAlertsTable component', () => {
  const props = {
    cautionaryAlerts: CAUTIONARY_ALERTS,
    query: ['VA', 'DAT', 'CIT'],
  }

  describe('when logged in as an operative and has NO queries', () => {
    it('should render properly: with NO highlithed codes', () => {
      const { asFragment } = render(
        <UserContext.Provider value={{ user: operative }}>
          <CautionaryAlertsTable cautionaryAlerts={props.cautionaryAlerts} />
        </UserContext.Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('when logged in as an operative and has queries', () => {
    it('should render properly: with highlithed codes', () => {
      const { asFragment } = render(
        <UserContext.Provider value={{ user: operative }}>
          <CautionaryAlertsTable
            cautionaryAlerts={props.cautionaryAlerts}
            query={props.query.cautContact}
          />
        </UserContext.Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
