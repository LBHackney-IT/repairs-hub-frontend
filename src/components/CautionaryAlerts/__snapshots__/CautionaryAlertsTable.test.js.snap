// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CautionaryAlertsTable component when logged in as an operative and has NO queries should render properly: with NO highlithed codes 1`] = `
<DocumentFragment>
  <table
    class="govuk-table lbh-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row lbh-body"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Code
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
        data-row-id="0"
      >
        <td
          class="govuk-table__cell"
        >
          DAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="1"
      >
        <td
          class="govuk-table__cell"
        >
          CFT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="2"
      >
        <td
          class="govuk-table__cell"
        >
          CIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="3"
      >
        <td
          class="govuk-table__cell"
        >
          CVT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="4"
      >
        <td
          class="govuk-table__cell"
        >
          OIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="5"
      >
        <td
          class="govuk-table__cell"
        >
          PAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="6"
      >
        <td
          class="govuk-table__cell"
        >
          SAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="7"
      >
        <td
          class="govuk-table__cell"
        >
          UET
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="8"
      >
        <td
          class="govuk-table__cell"
        >
          VAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Verbal Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="9"
      >
        <td
          class="govuk-table__cell"
        >
          COR
        </td>
        <td
          class="govuk-table__cell"
        >
          ASBO or injunction obtained
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="10"
      >
        <td
          class="govuk-table__cell"
        >
          DA
        </td>
        <td
          class="govuk-table__cell"
        >
          Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="11"
      >
        <td
          class="govuk-table__cell"
        >
          CX
        </td>
        <td
          class="govuk-table__cell"
        >
          Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="12"
      >
        <td
          class="govuk-table__cell"
        >
          DV
        </td>
        <td
          class="govuk-table__cell"
        >
          Domestic Violence
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="13"
      >
        <td
          class="govuk-table__cell"
        >
          CF
        </td>
        <td
          class="govuk-table__cell"
        >
          No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="14"
      >
        <td
          class="govuk-table__cell"
        >
          CI
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="15"
      >
        <td
          class="govuk-table__cell"
        >
          CV
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="16"
      >
        <td
          class="govuk-table__cell"
        >
          OI
        </td>
        <td
          class="govuk-table__cell"
        >
          Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="17"
      >
        <td
          class="govuk-table__cell"
        >
          PA
        </td>
        <td
          class="govuk-table__cell"
        >
          Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="18"
      >
        <td
          class="govuk-table__cell"
        >
          SA
        </td>
        <td
          class="govuk-table__cell"
        >
          Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="19"
      >
        <td
          class="govuk-table__cell"
        >
          SPR
        </td>
        <td
          class="govuk-table__cell"
        >
          Specific Requirements
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="20"
      >
        <td
          class="govuk-table__cell"
        >
          CXT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="21"
      >
        <td
          class="govuk-table__cell"
        >
          TRA
        </td>
        <td
          class="govuk-table__cell"
        >
          Translation Required
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="22"
      >
        <td
          class="govuk-table__cell"
        >
          UE
        </td>
        <td
          class="govuk-table__cell"
        >
          Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="23"
      >
        <td
          class="govuk-table__cell"
        >
          VA
        </td>
        <td
          class="govuk-table__cell"
        >
          Verbal Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="24"
      >
        <td
          class="govuk-table__cell"
        >
          APT
        </td>
        <td
          class="govuk-table__cell"
        >
          Adapted Property
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="25"
      >
        <td
          class="govuk-table__cell"
        >
          VOID
        </td>
        <td
          class="govuk-table__cell"
        >
          Property is VOID
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="26"
      >
        <td
          class="govuk-table__cell"
        >
          DIS
        </td>
        <td
          class="govuk-table__cell"
        >
          Property Under Disrepair
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="27"
      >
        <td
          class="govuk-table__cell"
        >
          TERM
        </td>
        <td
          class="govuk-table__cell"
        >
          Tenant has requested termination of tenancy please refer calls to housing officer
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CautionaryAlertsTable component when logged in as an operative and has queries should render properly: with highlithed codes 1`] = `
<DocumentFragment>
  <table
    class="govuk-table lbh-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row lbh-body"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Code
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
        data-row-id="0"
      >
        <td
          class="govuk-table__cell"
        >
          DAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="1"
      >
        <td
          class="govuk-table__cell"
        >
          CFT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="2"
      >
        <td
          class="govuk-table__cell"
        >
          CIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="3"
      >
        <td
          class="govuk-table__cell"
        >
          CVT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="4"
      >
        <td
          class="govuk-table__cell"
        >
          OIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="5"
      >
        <td
          class="govuk-table__cell"
        >
          PAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="6"
      >
        <td
          class="govuk-table__cell"
        >
          SAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="7"
      >
        <td
          class="govuk-table__cell"
        >
          UET
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="8"
      >
        <td
          class="govuk-table__cell"
        >
          VAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Verbal Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="9"
      >
        <td
          class="govuk-table__cell"
        >
          COR
        </td>
        <td
          class="govuk-table__cell"
        >
          ASBO or injunction obtained
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="10"
      >
        <td
          class="govuk-table__cell"
        >
          DA
        </td>
        <td
          class="govuk-table__cell"
        >
          Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="11"
      >
        <td
          class="govuk-table__cell"
        >
          CX
        </td>
        <td
          class="govuk-table__cell"
        >
          Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="12"
      >
        <td
          class="govuk-table__cell"
        >
          DV
        </td>
        <td
          class="govuk-table__cell"
        >
          Domestic Violence
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="13"
      >
        <td
          class="govuk-table__cell"
        >
          CF
        </td>
        <td
          class="govuk-table__cell"
        >
          No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="14"
      >
        <td
          class="govuk-table__cell"
        >
          CI
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="15"
      >
        <td
          class="govuk-table__cell"
        >
          CV
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="16"
      >
        <td
          class="govuk-table__cell"
        >
          OI
        </td>
        <td
          class="govuk-table__cell"
        >
          Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="17"
      >
        <td
          class="govuk-table__cell"
        >
          PA
        </td>
        <td
          class="govuk-table__cell"
        >
          Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="18"
      >
        <td
          class="govuk-table__cell"
        >
          SA
        </td>
        <td
          class="govuk-table__cell"
        >
          Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="19"
      >
        <td
          class="govuk-table__cell"
        >
          SPR
        </td>
        <td
          class="govuk-table__cell"
        >
          Specific Requirements
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="20"
      >
        <td
          class="govuk-table__cell"
        >
          CXT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="21"
      >
        <td
          class="govuk-table__cell"
        >
          TRA
        </td>
        <td
          class="govuk-table__cell"
        >
          Translation Required
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="22"
      >
        <td
          class="govuk-table__cell"
        >
          UE
        </td>
        <td
          class="govuk-table__cell"
        >
          Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="23"
      >
        <td
          class="govuk-table__cell"
        >
          VA
        </td>
        <td
          class="govuk-table__cell"
        >
          Verbal Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="24"
      >
        <td
          class="govuk-table__cell"
        >
          APT
        </td>
        <td
          class="govuk-table__cell"
        >
          Adapted Property
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="25"
      >
        <td
          class="govuk-table__cell"
        >
          VOID
        </td>
        <td
          class="govuk-table__cell"
        >
          Property is VOID
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="26"
      >
        <td
          class="govuk-table__cell"
        >
          DIS
        </td>
        <td
          class="govuk-table__cell"
        >
          Property Under Disrepair
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="27"
      >
        <td
          class="govuk-table__cell"
        >
          TERM
        </td>
        <td
          class="govuk-table__cell"
        >
          Tenant has requested termination of tenancy please refer calls to housing officer
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
